addons/sms/wizard/sms_composer.py

```diff
@@ -230,7 +230,7 @@ def _action_send_sms_comment_single(self, records=None):
         # on the numbers in the database.
         records = records if records is not None else self._get_records()
         records.ensure_one()
-        if self.recipient_single_number_itf and self.recipient_single_number_itf != self.recipient_single_number:
+        if self.number_field_name in self._fields and self.recipient_single_number_itf and self.recipient_single_number_itf != self.recipient_single_number:
```

Comment:
If you do this, this will always be false because self is the composer.

Make sure to write a test to show why it failed.